packages/app/src/plugins/cost-insights/ExampleCostInsightsClient.ts

```diff
@@ -116,19 +117,33 @@ export class ExampleCostInsightsClient implements CostInsightsApi {
     return projects;
   }
 
-  async getGroupDailyCost(
-    group: string,
+  async getMetricData(
     metric: string | null,
```

Comment:
Nice catch. No, it can't but then `getMetricData` would not accept the type signature for `Metric.kind`.

We guard against null when calling `getMetricData` since `dailyCost` is considered a null metric and we already get dailyCosts from `getProjectDailyCost` and `getGroupDailyCost`. 

The simple solution here would be to simply change the type on the arg but maybe this means we need to retype `Metric` ?

Reply:
Hm, doesn't `getMetricData` need to accept the type for `Metric.compare` ? We call `getMetricData` w/ `pageFilters.metric` which comes from `metric.compare` where the `metric.kind` is `null` in `useFilters.tsx`
```    
const compared = config.metrics.find(m => m.kind === null)?.compare;
setPageFilters({ ...initialState, metric: compared || null }); 
```